lib/bitvec.cpp

```diff
@@ -145,3 +146,34 @@ bool bitvec::is_contiguous() const {
         return false;
     return max().index() - min().index() + 1 == popcount();
 }
+
+bitvec bitvec::rotate_helper(size_t start_bit, size_t rotation_idx, size_t end_bit) const {
+    BUG_CHECK(start_bit <= rotation_idx && rotation_idx < end_bit, "Invalid rotation on bitvec, as "
+              "rotation_idx does not fall between start_bit and end_bit");
+    bitvec rot_mask(start_bit, end_bit - start_bit);
+    bitvec rotation_section = *this & rot_mask; 
+    int down_shift = rotation_idx - start_bit;
+    int up_shift = (end_bit - start_bit) - down_shift;
+    bitvec rv;
+    rv = (rotation_section >> down_shift) | (rotation_section << up_shift);
+    return rv & rot_mask; 
+}
+
+/**
+ * Designed to imitate the std::rotate/std::rotate_copy function for vectors.  Return a bitvec
```

Comment:
Note that std::rotate uses big-endian indexing, so a left shift is shifting down (towards index 0), while we use little endian indexing for bitvecs, so a left shift is shifting up (away from index 0).  Very confusing.

Reply:
Would you prefer a different implementation?  I can make the rotate function shift the other way.  I was just going off of a base implementation.

Reply:
or I can rename the functions to rotate_right/rotate_copy_right if you'd prefer that as well, for much clearer naming.

Reply:
I think its fine -- might just want to note the right/left difference in the comment.